elastic/kibana
# [Infra] Add custom dashboard tracking event

Closes #182859 
## Summary

This PR adds a tracking event `Asset Dashboard Loaded` when the custom dashboard is loaded on the dashboards tab.

So to understand the usage of the filters we want to track: 
- when the page loads with the dashboard tab open
- If the dashboard filters change or a different dashboard is selected 
- when the dashboard is initially loaded (and the filters applied)
We don't want to track:
- If the dashboard is not changed, in case: 
   - Edit action is canceled or the same filters are saved 
   - the same dashboard is selected 
   - switching to a different tab and back to the dashboards tab (showing the same dashboard)

## Testing
Go to the asset details flyout/ asset details page and select the dashboards tab
- Add a dashboard with a filter ( Check tracking )
  ![image](https://github.com/elastic/kibana/assets/14139027/690ba626-714c-4325-8435-d43f5c914f4d)

- Add a dashboard without a filter  ( Check tracking )
  ![image](https://github.com/elastic/kibana/assets/14139027/7f4c009a-3c1b-4301-961d-e6e2407f5e75)

- Switch dashboards ( Check tracking )
- Edit a dashboard ( Check tracking )



https://github.com/elastic/kibana/assets/14139027/b43f3e88-25a8-46d4-be88-087917f577db

- Should not track: 

https://github.com/elastic/kibana/assets/14139027/34d93849-fe0d-4643-ac7e-d8de0b74820c
